src/JsonSchema/TypeFactory.php

```diff
@@ -50,14 +63,40 @@ public function setSchemaFactory(SchemaFactoryInterface $schemaFactory): void
     public function getType(Type $type, string $format = 'json', ?bool $readableLink = null, ?array $serializerContext = null, Schema $schema = null): array
     {
         if ($type->isCollection()) {
-            $subType = new Type($type->getBuiltinType(), $type->isNullable(), $type->getClassName(), false);
-
-            return [
-                'type' => 'array',
-                'items' => $this->getType($subType, $format, $readableLink, $serializerContext, $schema),
-            ];
+            $keyType = $type->getCollectionKeyType();
+            $subType = $type->getCollectionValueType()
+                ?? new Type($type->getBuiltinType(), false, $type->getClassName(), false);
+
+            if (null !== $keyType && Type::BUILTIN_TYPE_STRING === $keyType->getBuiltinType()) {
+                return $this->addNullabilityToTypeDefinition(
+                    [
+                        'type' => 'object',
```

Comment:
Can't it be an actual collection of strings, instead of IRIs? (If I understand the code correctly.)

Reply:
Don't we need to check the value type too?

Reply:
If it has string keys, it is not an `array` in JSON-Schema. `array` and `map` are very clear concepts in other languages: not in PHP (we do everything in one shot, and that's the cause of a lot of confusion)

Reply:
Uhh, yes. That flew way past my head. :laughing: